### PR TITLE
Adds explanations for all minikube error codes

### LINKED ISSUE #11555: Add explanations for all minikube error codes
We now populate documentation for all error/exit codes at https://minikube.sigs.k8s.io/docs/contrib/errorcodes/, but most of them don't have a comment explaining what they are. We should add those.

### PR DESCRIPTION
**Pull request by sharmarajdaksh**
This PR adds basic descriptions for all of the used error codes, and thus fixes #11555

### CODE REVIEW

**Comment by sharmarajdaksh** [comment]
@sharifelgamal let me know if this looks good!

File: pkg/minikube/reason/reason.go [hunk 1]
```diff
@@ -64,54 +64,98 @@ var (
...
+	InternalReservedProfile = Kind{ID: "MK_RESERVED_PROFILE", ExitCode: ExProgramConflict}
+	// minkube failed to generate script to set or unset minikube-env
+	InternalEnvScript = Kind{ID: "MK_ENV_SCRIPT", ExitCode: ExProgramError}
+	// minikube failed to detect the shell in use
+	InternalShellDetect = Kind{ID: "MK_SHELL_DETECT", ExitCode: ExProgramError}
+	// minikube failed to output JSON-formatted minikube status
+	InternalStatusJSON = Kind{ID: "MK_STATUS_JSON", ExitCode: ExProgramError}
+	// minikube failed to output minikube status text
+	InternalStatusText = Kind{ID: "MK_STATUS_TEXT", ExitCode: ExProgramError}
+	// minikube failed to generate script to deactive minikube docker-env
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:139]: "to deactivate" instead of "to deactive"

File: pkg/minikube/reason/reason.go [hunk 2]
```diff
@@ -64,54 +64,98 @@ var (
...
+	InternalUnsetScript = Kind{ID: "MK_UNSET_SCRIPT", ExitCode: ExProgramError}
+	// minikube failed to create view template for displaying current config
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:141]: Flip this comment with the one on the next line. InternalViewExec is about executing the view template, and InternalViewTmpl is about creating the view template.

File: pkg/minikube/reason/reason.go [hunk 3]
```diff
@@ -64,54 +64,98 @@ var (
...
+	InternalViewExec = Kind{ID: "MK_VIEW_EXEC", ExitCode: ExProgramError}
+	// minikube failed to execute view template for displaying current config
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:143]: "execute (fill in values)" instead of "execute"

File: pkg/minikube/reason/reason.go [hunk 4]
```diff
@@ -64,54 +64,98 @@ var (
...
+	InternalViewTmpl = Kind{ID: "MK_VIEW_TMPL", ExitCode: ExProgramError}
+	// minikube failed to marshal a YAML object
+	InternalYamlMarshal = Kind{ID: "MK_YAML_MARSHAL", ExitCode: ExProgramError}
+	// minikube could not locate credentials needed to utilize an appropriate service, such as GCP
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:147]: "e.g. GCP" instead of "such as GCP"

File: pkg/minikube/reason/reason.go [hunk 5]
```diff
@@ -64,54 +64,98 @@ var (
...
+	InternalCredsNotFound = Kind{ID: "MK_CREDENTIALS_NOT_FOUND", ExitCode: ExProgramNotFound, Style: style.Shrug}
+	// minikube was passed service credentials when they were not needed, such as when using the GCP Auth addon when running in GCE
+	InternalCredsNotNeeded = Kind{ID: "MK_CREDENTIALS_NOT_NEEDED", ExitCode: ExProgramNotFound, Style: style.Shrug}
+	// minikube found an invalid semver string for kubernetes in the minikube constants
+	InternalSemverParse = Kind{ID: "MK_SEMVER_PARSE", ExitCode: ExProgramError}
+	// minikube was unable to daemonize the minikube procecess
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:153]: typo: "process"

File: pkg/minikube/reason/reason.go [hunk 6]
```diff
@@ -64,54 +64,98 @@ var (
...
+	DaemonizeError = Kind{ID: "MK_DAEMONIZE", ExitCode: ExProgramError}
 
-	RsrcInsufficientCores             = Kind{ID: "RSRC_INSUFFICIENT_CORES", ExitCode: ExInsufficientCores, Style: style.UnmetRequirement}
+	// insufficient cores available for use by minikube and kubernetes
+	RsrcInsufficientCores = Kind{ID: "RSRC_INSUFFICIENT_CORES", ExitCode: ExInsufficientCores, Style: style.UnmetRequirement}
+	// insufficient cores available for use by docker desktop on mac
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:158]: "Docker Desktop on Mac"

File: pkg/minikube/reason/reason.go [hunk 7]
```diff
@@ -124,6 +168,7 @@ var (
 		URL:   "https://docs.docker.com/docker-for-mac/#resources",
 	}
 
+	// insufficient cores available for use by docker desktop on windows
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:171]: "Docker Desktop on Windows"

File: pkg/minikube/reason/reason.go [hunk 8]
```diff
@@ -136,9 +181,13 @@ var (
...
-	RsrcInsufficientReqMemory           = Kind{ID: "RSRC_INSUFFICIENT_REQ_MEMORY", ExitCode: ExInsufficientMemory, Style: style.UnmetRequirement}
-	RsrcInsufficientSysMemory           = Kind{ID: "RSRC_INSUFFICIENT_SYS_MEMORY", ExitCode: ExInsufficientMemory, Style: style.UnmetRequirement}
-	RsrcInsufficientContainerMemory     = Kind{ID: "RSRC_INSUFFICIENT_CONTAINER_MEMORY", ExitCode: ExInsufficientMemory, Style: style.UnmetRequirement}
+	// insufficient memory (less than the recommended minimum) allocated to minikube
+	RsrcInsufficientReqMemory = Kind{ID: "RSRC_INSUFFICIENT_REQ_MEMORY", ExitCode: ExInsufficientMemory, Style: style.UnmetRequirement}
+	// insufficient memory (less than the recommended minimum) available on the system running minikube
+	RsrcInsufficientSysMemory = Kind{ID: "RSRC_INSUFFICIENT_SYS_MEMORY", ExitCode: ExInsufficientMemory, Style: style.UnmetRequirement}
+	// insufficient memory available for the driver in use by minikube
+	RsrcInsufficientContainerMemory = Kind{ID: "RSRC_INSUFFICIENT_CONTAINER_MEMORY", ExitCode: ExInsufficientMemory, Style: style.UnmetRequirement}
+	// insufficient memory available to docker desktop on windows
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:190]: "Docker Desktop on Windows"

File: pkg/minikube/reason/reason.go [hunk 9]
```diff
@@ -150,6 +199,7 @@ var (
 		URL:   "https://docs.docker.com/docker-for-windows/#resources",
 		Style: style.UnmetRequirement,
 	}
+	// insufficient memory available to docker desktop on mac
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:202]: "Docker Desktop on Mac"

File: pkg/minikube/reason/reason.go [hunk 10]
```diff
@@ -183,120 +235,211 @@ var (
...
+	// insufficient disk storage available for running minikube and kubernetes
 	RsrcInsufficientStorage = Kind{ID: "RSRC_INSUFFICIENT_STORAGE", ExitCode: ExInsufficientStorage, Style: style.UnmetRequirement}
 
-	HostHomeMkdir      = Kind{ID: "HOST_HOME_MKDIR", ExitCode: ExHostPermission}
-	HostHomeChown      = Kind{ID: "HOST_HOME_CHOWN", ExitCode: ExHostPermission}
-	HostBrowser        = Kind{ID: "HOST_BROWSER", ExitCode: ExHostError}
-	HostConfigLoad     = Kind{ID: "HOST_CONFIG_LOAD", ExitCode: ExHostConfig}
+	// minikube could not create the minikube directory
+	HostHomeMkdir = Kind{ID: "HOST_HOME_MKDIR", ExitCode: ExHostPermission}
+	// minikube could change permissions for the minikube directory
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:243]: "could not"

File: pkg/minikube/reason/reason.go [hunk 11]
```diff
@@ -183,120 +235,211 @@ var (
...
-	HostPathStat            = Kind{ID: "HOST_PATH_STAT", ExitCode: ExHostError}
-	HostPurge               = Kind{ID: "HOST_PURGE", ExitCode: ExHostError}
-	HostSaveProfile         = Kind{ID: "HOST_SAVE_PROFILE", ExitCode: ExHostConfig}
+	// minikube failed to launch a kubectl proxy
+	HostKubectlProxy = Kind{ID: "HOST_KUBECTL_PROXY", ExitCode: ExHostError}
+	// minikube failed to write mount pid
+	HostMountPid = Kind{ID: "HOST_MOUNT_PID", ExitCode: ExHostError}
+	// minikube was passed a path to a host directory that does not exist
+	HostPathMissing = Kind{ID: "HOST_PATH_MISSING", ExitCode: ExHostNotFound}
+	// minikube failed to stat a directory path
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:275]: "failed to get info about a directory path" maybe?

File: pkg/minikube/reason/reason.go [hunk 12]
```diff
@@ -183,120 +235,211 @@ var (
...
+	GuestNodeProvision = Kind{ID: "GUEST_NODE_PROVISION", ExitCode: ExGuestError}
+	// minikube failed to retrieve information for a cluster node
+	GuestNodeRetrieve = Kind{ID: "GUEST_NODE_RETRIEVE", ExitCode: ExGuestNotFound}
+	// minikube failed to startup a cluster node
+	GuestNodeStart = Kind{ID: "GUEST_NODE_START", ExitCode: ExGuestError}
+	// minikube failed to pause the cluster process
+	GuestPause = Kind{ID: "GUEST_PAUSE", ExitCode: ExGuestError}
+	// minikube failed to delete a machine profile directory
+	GuestProfileDeletion = Kind{ID: "GUEST_PROFILE_DELETION", ExitCode: ExGuestError}
+	// minikube failed while attempting to provision the  host
```
Comment (andriyDev) [inline pkg/minikube/reason/reason.go:354]: I think this should be "provision the guest", @sharifelgamal right? The error message [here](https://github.com/andriyDev/minikube/blob/4f114dafb9289cf966b914d8d86e26c31918310a/cmd/minikube/cmd/start.go#L1516) is wrong I think, this should probably say "error provisioning guest". We should fix that in this PR.

**Comment by sharmarajdaksh** [comment]
Thanks for the review @andriyDev. I'll make the changes you mentioned

File: pkg/minikube/reason/reason.go (hunk 12 above)
Comment (sharmarajdaksh) [reply pkg/minikube/reason/reason.go:354]: Hey @andriyDev, @sharifelgamal 
I've done updating the rest of comments mentioned by andriy, any update/help on this? Should I change it here to Guest, as well as in the location you mentioned?

Comment (sharifelgamal) [reply pkg/minikube/reason/reason.go:354]: so yeah, in this case, guest and host are being used interchangeably. what it means is that we failed to start the VM or container that minikube lives in. I agree with @andriyDev in this case, change the wording here and in the error message itself.

**Comment by sharmarajdaksh** [comment]
@andriyDev @sharifelgamal I've updated the PR with all of the changes you mentioned

File: pkg/minikube/reason/reason.go [hunk 13]
```diff
@@ -64,54 +64,98 @@ var (
...
-	InternalViewExec         = Kind{ID: "MK_VIEW_EXEC", ExitCode: ExProgramError}
-	InternalViewTmpl         = Kind{ID: "MK_VIEW_TMPL", ExitCode: ExProgramError}
-	InternalYamlMarshal      = Kind{ID: "MK_YAML_MARSHAL", ExitCode: ExProgramError}
-	InternalCredsNotFound    = Kind{ID: "MK_CREDENTIALS_NOT_FOUND", ExitCode: ExProgramNotFound, Style: style.Shrug}
-	InternalCredsNotNeeded   = Kind{ID: "MK_CREDENTIALS_NOT_NEEDED", ExitCode: ExProgramNotFound, Style: style.Shrug}
-	InternalSemverParse      = Kind{ID: "MK_SEMVER_PARSE", ExitCode: ExProgramError}
-	DaemonizeError           = Kind{ID: "MK_DAEMONIZE", ExitCode: ExProgramError}
+	// minikube failed to list some configuration data
+	InternalListConfig = Kind{ID: "MK_LIST_CONFIG", ExitCode: ExProgramError}
+	// minikube failed to write logs to stdout (currently not in use)
```
Comment (sharifelgamal) [inline pkg/minikube/reason/reason.go:119]: stderr, not stdout

**Comment by sharifelgamal** [comment]
This looks great @sharmarajdaksh, I'll merge it once the last comment is addressed.

**Pull request merged**